urllib3/util/retry.py

```diff
@@ -290,5 +290,31 @@ def __repr__(self):
                     cls=type(self), self=self)
 
 
+def handle_redirect(response, requester, retries, method, url, *args, **kwargs):
+    """
+    Abstraction of the redirect process so that we can access it from
+    multidomain levels in addition to from the :class:`.ConnectionPool`.
+    """
+    # urljoin will give us either a full URL or just a path, based on what
+    # the original URL is, making it appropriate for the ConnectionPool
+    # level as well as the PoolManager or SessionManager
+    redirect_location = urljoin(url, response.get_redirect_location())
```

Comment:
Hrrrm. Does this actually work? Location headers can come in a wide variety of forms and styles, and in particular can come without a scheme.

Additionally, my reading of this PR is that it *changes* the redirect logic currently possessed by urllib3, which previously didn't do this urljoin. Why the change?